### PR TITLE
Stdlib doc changes

### PR DESCRIPTION
**Pull request by amartini51**
Corrections to documentation comments.

Resolves rdar://problem/59833804&58855960&58717942

See also the [PR](https://github.com/apple/swift/pull/29435) to incorporate the same changes into the Swift 5.2 branch.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

File: stdlib/public/core/OptionSet.swift [hunk 1]
```diff
@@ -278,7 +278,7 @@ extension OptionSet where Element == Self {
   ///     print(expressOption == .express)
   ///     // Prints "false"
   ///     print(expressOption == .secondDay)
-  ///     // Prints "true"
+  ///     // Prints "false"
```
Comment (lorentey) [inline stdlib/public/core/OptionSet.swift]: This change isn't right -- the documentation actually described the expected behavior here, even though the code got it wrong in previous versions. (The return value is prescribed by `SetAlgebra`). 

The stdlib used to return the wrong value, but this was fixed on master -- see https://github.com/apple/swift/pull/28378.

**Review by lorentey — CHANGES_REQUESTED** [review]

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

File: stdlib/public/core/OptionSet.swift (hunk 1 above)
Comment (amartini51) [reply stdlib/public/core/OptionSet.swift]: Thanks. Removed this change/commit.

**Comment by amartini51** [comment]
Picked up one more correction and dropped a commit where the docs were actually right.  Still need to fix the failing test where SourceKit has a hardcoded expectation of what some Array docs look like.

**Review by lorentey — APPROVED** [review]

**Pull request merged**